openassessment/workflow/models.py

```diff
@@ -640,10 +652,120 @@ def get_by_team_submission_uuid(cls, team_submission_uuid):
+                self.uuid,
+                self.steps.all()
+            )
+            logger.error(err_msg)
+            raise AssessmentWorkflowInternalError(err_msg)
+        step = self.steps.first()
+        if step.name != TeamAssessmentWorkflow.STATUS.teams:
+            err_msg = 'Team Assessment Workflow {} has a "{}" step rather than a teams step'.format(
+                self.uuid,
+                step.name
+            )
+            logger.error(err_msg)
+            raise AssessmentWorkflowInternalError(err_msg)
+        return [step]
+
     def update_from_assessments(self):  # pylint: disable=arguments-differ
-        """ Update status from the assessments. For teams, only the staff assessment """
-        # TODO - Implement in https://openedx.atlassian.net/browse/EDUCATOR-4986
-        pass
+        """
+        Update the workflow with potential new scores from assessments.
+        """
+        if self.status == self.STATUS.cancelled:
+            return
+
+        team_staff_step = self._team_staff_step
+        team_staff_api = team_staff_step.api()
+        new_score = team_staff_api.get_score(self.team_submission_uuid, self.REQUIREMENTS)
+        if new_score:
+            # new_score is just the most recent team score, it may already be recorded in sub_api
+            old_score = sub_api.get_latest_score_for_submission(self.submission_uuid)
+            if (
+                    # Does a prior score exist?  Do the points earned match?
+                    not old_score or self.STAFF_ANNOTATION_TYPE not in [
+                        annotation['annotation_type'] for annotation in old_score['annotations']
+                    ] or old_score['points_earned'] != new_score['points_earned']
+            ):
+                # Set the team staff score using team submissions api, and log that fact
+                self._set_team_staff_score(new_score)
+                self.save()
+                logger.info((
+                    u"Team Workflow for team submission UUID {uuid} has updated score using team staff assessment."
+                ).format(uuid=self.team_submission_uuid))
+
+                team_staff_step.assessment_completed_at = now()
+                team_staff_step.save()
+
+            team_staff_step.update(self.team_submission_uuid, self.REQUIREMENTS)
+            self.status = self.STATUS.done
+            self.save()
+
+    def _set_team_staff_score(self, score):
+        reason = "A staff member has defined the score for this submission"
+        sub_team_api.set_score(
+            self.team_submission_uuid,
+            score["points_earned"],
+            score["points_possible"],
+            annotation_creator=score["staff_id"],
+            annotation_type=self.STAFF_ANNOTATION_TYPE,
+            annotation_reason=reason
+        )
+
+    @property
+    def identifying_uuid(self):
+        """
+        Returns the primary identifying uuid for this workflow.
```

Comment:
I would change the doc string as follows:
Overwrites AssessmentWorkflow.identifying_uuid to return team_submission_uuid